### ISSUE TITLE
static curl with static c-ares on MacOS not linking correctly

### ISSUE THREAD

**Issue by TvdW** [curl/curl#6057]
### I did this

```
# c-ares
./configure --prefix <stuff> --disable-shared --enable-static --with-pic --enable-nonblocking
make install

# curl
./configure --prefix <stuff> --with-zlib --with-nghttp2=<stuff> \
    --with-brotli=<stuff> --enable-ares=<stuff> \
    --disable-shared --enable-static --with-pic \
    --disable-threaded-resolver --disable-pthreads
```

This fails with `configure: error: c-ares library defective or too old`.

config.log:
```
configure:42592: checking that c-ares is good and recent enough
configure:42619: gcc -o conftest -mmacosx-version-min=10.8 -Qunused-arguments -Os -Werror=partial-availability   -I<stuff> -L<stuff> conftest.c -lcares -lnghttp2 -lldap -lbrotlidec -lbrotlidec -lz  >&5
Undefined symbols for architecture x86_64:
  "_res_9_getservers", referenced from:
      _ares_init_options in libcares.a(libcares_la-ares_init.o)
  "_res_9_ndestroy", referenced from:
      _ares_init_options in libcares.a(libcares_la-ares_init.o)
  "_res_9_ninit", referenced from:
      _ares_init_options in libcares.a(libcares_la-ares_init.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Those symbols come from libresolv, and adding `-lresolv` to LDFLAGS does fix the build.

libcares.pc looks like this:
```
prefix=<stuff>
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/includeName: c-ares
URL: http://daniel.haxx.se/projects/c-ares/
Description: asynchronous DNS lookup library
Version: 1.16.1
Requires:
Requires.private:
Cflags: -I${includedir} -DCARES_STATICLIB
Libs: -L${libdir} -lcares
Libs.private: -lresolv
```

I considered reporting this issue over at https://github.com/c-ares/c-ares/ but I'm not sure whether this happens because curl doesn't pick up the private libs, or that it should be part of the regular libs section. Since the committers to the repos largely overlap I'm filing the bug report with curl for now, since that's the `configure` that breaks.

### tldr

static c-ares build has a dependency on libresolv on MacOS, that is not picked up by curl, causing the build to fail.

### curl/libcurl version

This would be on 7.71.1. This isn't happening on my machine (don't have a Mac) so hard to test with latest git version. c-ares 1.16.1.

### operating system

MacOS 10.15.7

**Comment by bagder** [comment]
This is [a known issue](https://curl.haxx.se/docs/knownbugs.html#Build_with_statically_built_depe).

It's the ever returning issue with static linking and the inability to know what other dependencies to link with. The configure script doesn't even know that it only has a static c-ares library to choose. The curl configure simply doesn't know what sub-dependencies c-ares has so when it fails to link it fails.

The script *could* of course try adding various libraries in case of failure or ask `pkg-config` for the `--static` list as one alternative, but experience has shown that this kind of chasing dependencies is rather futile and will never be 100% - especially not over time as dependencies shift.

Because of this, we usually ask users to provide the exact extra deps with `LDFLAGS=` when linking with static libraries.

**Comment by TvdW** [comment]
Yeah, fair enough, I did see it in the known issues section but wasn't sure whether this was the same problem, or a missing dependency in the pkg-config of c-ares (I'm no C developer, don't know if `-lresolv` is supposed to be in there or not...)

Feel free to close this in that case, since it looks like there isn't much that can be done here.

**Comment by bagder** [comment]
Thanks, yes I'll stuff this under the "static builds are really complicated" label and close.
